7_examples/20_annotations.tex

```diff
@@ -0,0 +1,112 @@
+       q_{t_1}(2)=q_{t_2}(2)=q_{t_3}(2)=-x,\\
+       q_{t_4}(0)=-x+1,\\
+       q_{t_4}(1)=-\frac{x}{2}-1,\\
+       q_{t_4}(2)=-1,
+    \end{aligned}
+\end{equation}
+і якостями ребер 
+\begin{equation}
+    g_{tt'}(k,k')=-\left(\frac{x}{2}+1\right)\cdot\llbracket k\neq k'\rrbracket,
+\end{equation}
+для яких задача не є супермодулярною.
+
+Запустимо процедуру субградієнтного спуску. Виберемо початкову репараметризацію
+\begin{equation}
+    \varphi_{tt'}(k)=0,\forall k\in K, t\in T, t'\in N_t,
+\end{equation}
+і крок субградієнтного спуску 
+\begin{equation}
+    \gamma_i=\left(\frac{x}{2}+1\right)/i.
+\end{equation}
+Виконаємо 2 ітерації субградієнтного спуску й отримаємо значення репараметризації 
+\begin{equation}
+    \begin{aligned}
+       \varphi_{t_3t_4}(0)=\left(\frac{x}{2}+1\right)/2, \varphi_{t_3t_4}(2)=-\left(\frac{x}{2}+1\right)/2,\\
+       \varphi_{t_2t_4}(0)=\left(\frac{x}{2}+1\right)/2, \varphi_{t_2t_4}(2)=-\left(\frac{x}{2}+1\right)/2,\\
+       \varphi_{t_4t_3}(0)=-\left(\frac{x}{2}+1\right)/2, \varphi_{t_4t_3}(2)=\left(\frac{x}{2}+1\right)/2,\\
+       \varphi_{t_4t_2}(0)=-\left(\frac{x}{2}+1\right)/2, \varphi_{t_4t_2}(2)=\left(\frac{x}{2}+1\right)/2.\\
+    \end{aligned}
+\end{equation}
+Всі інші значення репараметризації $\varphi$ не змінилися. За такої репараметризації і 
+\begin{equation}
+    \varepsilon\leq \frac{1}{|T|+2\cdot|\Gamma|+1}\approx 0.077,
+\end{equation}
+задача є $\varepsilon$-узгодженою.
+Значення цільової функції 
+\begin{equation}
+    E(q,g,\varphi)=-x+1.
+\end{equation}
+У кожному об'єкті всі мітки мають скінченну вагу, тому другий крок алгоритму пропустимо.
+Обхід об'єктів графу здійснюємо у порядку $(t_1,t_2,t_3,t_4)$, а обхід міток в 
+об'єкті виконуємо у порядку $(0,1,2)$. Нехай $K_1=\{0\}$, $K_2=\{1,2\}$. Зафіксуємо мітку $0$ в 
+об'єкті $t_1$ та заборонимо інші мітки у цьому об'єкті
+\begin{equation}
+    \begin{aligned}
+    q'_{t_1}(0)=0,q'_{t_1}(1)=\infty,q'_{t_1}(2)=\infty,\\
+    q'_t(k)=q_t(k),\forall t\in \{t_2,t_3,t_4\}, \forall k\in K.
+    \end{aligned}
+\end{equation}
+Використаємо значення $\varphi$, отримані на попередньому кроці. Одразу маємо 
+$\varepsilon$-узгоджений набір вершин та ребер, а також 
+\begin{equation}
+    E(q',g,\varphi)=E(q,g,\varphi)=-x+1.
+\end{equation}
+В об'єкті $t_1$ залишимо мітку $0$, тобто $k^*_{t_1}=0$. Проведемо аналогічну процедуру 
+для всіх інших об'єктів. У результаті отримаємо оптимальну розмітку $k^*$.
+\begin{equation}
+    k^*(t_1)=k^*(t_2)=k^*(t_3)=k^*(t_4)=0,
+\end{equation}
+з
+\begin{equation}
+    G(k^*)=-x+1>E(q,g,\varphi)-1=-x.
+\end{equation}
+
+\section{Результати експериментів}
+
+Як приклад застосування алгоритму було використано задачу знешумлення бінарного зображення 
+\cite{Boykov,Boykov_2,comp_vision,Greig_port}. Зображення з шумом отримується шляхом додавання сильного
+гаусового шуму до початкового зображення (без шуму). 
+Нехай $T$ --- множина пікселів зображення, $K=\{0,1\}$ --- множина міток, де $0$ --- чорний колір 
```

Comment:
Try using `~` (non-breaking space) to avoid unwanted line breaks
```suggestion
Нехай $T$~---~множина пікселів зображення, $K=\{0,1\}$~---~множина міток, де $0$~---~чорний колір 
```